app/controllers/foreman_salt/minions_controller.rb

```diff
@@ -6,14 +6,21 @@ class MinionsController < ::ForemanSalt::ApplicationController
     add_smart_proxy_filters :node, :features => 'Salt'
 
     def node
-      enc = {}
+      enc = {'foreman' => {}}
       env = @minion.salt_environment.blank? ? 'base' : @minion.salt_environment.name
-      enc['classes'] = @minion.salt_modules_for_enc
+      enc['foreman']['classes'] = @minion.salt_modules_for_enc
 
       pillars = @minion.info['parameters']
-      enc['parameters'] = Setting[:salt_namespace_pillars] ? { 'foreman' => pillars } : pillars
+      enc['foreman']['parameters'] = Setting[:salt_namespace_pillars] ? { 'foreman' => pillars } : pillars
 
-      enc['environment'] = env
+      enc['foreman']['parameters'].each do |key, value|
+        if (value.is_a? String) and value.start_with?("---\r\n")
```

Comment:
Just noticed that if a CR is added after the '---' there is \r\n.
Of course we can just check '---'